tools/superdsm/run-superdsm.py

```diff
@@ -0,0 +1,88 @@
+"""
+Copyright 2023 Leonid Kostrykin, Biomedical Computer Vision Group, Heidelberg University.
+
+Distributed under the MIT license.
+See file LICENSE for detail or copy at https://opensource.org/licenses/MIT
+
+"""
+
+import argparse
+import imghdr
+import pathlib
+import shutil
+import tempfile
+
+import ray
+import superdsm.automation
+import superdsm.io
+import superdsm.render
+
+
+# It was not possible to get ray to work in conjunction with planemo, though the reason remains

```

Comment:
Oo, can you give me a commit that is not patching out anything and is an ideal script? Galaxy should not do anything to the job ... or lets say the script that Galaxy generates can be executed without Galaxy and we can see where the memory leak is.

Reply:
@bgruening Sure, please have a look at #62. I'm not sure whether there really is a memory leak, I rather suspect that the memory constraints enforced by planemo/Galaxy are too strict for ray.

Reply:
Galaxy/planemo do not have that restriction in test mode. I expect more a problem with the local job runner in Galaxy and the threading in ray. Damn, need to think about it. Never saw this before.

Reply:
I agree that getting ray to work properly here would be good. Maybe it's worth noting that a minimal example I created of using ray ran without any issues with `planemo test`. Would it help if I created a minimal example which fails similarly to the here observed behavior? @bgruening  

Still, this sounds like an issue which will take some more time before it can be resolved. Thus I would prefer to first implement this tool with ray "patched out" and then fix the issues in the next iteration. Do you think we can proceed like this? In that case I will try to figure out why the [recent build](https://github.com/BMCV/galaxy-image-analysis/actions/runs/5326885435) failed.

Reply:
So I created a minimal example in #63 which reproduces the issue:
```python
import numpy as np
import ray
import scipy.ndimage as ndi

if __name__ == "__main__":
    ray.init(num_cpus=1, log_to_driver=True)
    img = np.zeros((1344, 1024))

    for _ in range(3):
        ndi.gaussian_laplace(img, 200)

    ray.put(img)
```

The example fails with the same `Broken pipe` error in `ray.put`. The test passes if the code is reduced any further. Any idea what might be causing that behavior?

Reply:
#62 passed after I downgraded ray from 2.3 to 1.6, which is a better workaround than the patching used in this PR. Do you think that #62 can be merged now? @bgruening